src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+
+        foreach ($domains as $domain) {
+            $fileId = $this->getFileIdByDomain($fileList, $domain);
+
+            if (!$fileId) {
+                continue;
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $response = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $response = $this->downloadSourceFile($fileId);
+                }
+
+                $responses[] = [
+                    'response' => $response,
+                    'locale' => $locale,
+                    'domain' => $domain,
+                ];
+            }
+        }
+
+        foreach ($responses as $responseData) {
+            /** @var ResponseInterface $response */
+            $response = $responseData['response'];
+
+            if (204 === $response->getStatusCode()) {
+                $this->logger->error(sprintf('No content in exported file. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            if (200 !== $response->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to export file. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            $exportResponse = $this->filesDownloader->request('GET', $response->toArray()['data']['url']);
+
+            if (200 !== $exportResponse->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to download file content. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            $translatorBag->addCatalogue($this->loader->load($exportResponse->getContent(), $responseData['locale'], $responseData['domain']));
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $fileList = $this->getFileList();
+        $responses = [];
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            if ($catalogue->getLocale() !== $this->defaultLocale) {
+                continue;
+            }
+
+            foreach ($catalogue->getDomains() as $domain) {
+                $fileId = $this->getFileIdByDomain($fileList, $domain);
+
+                if (!$fileId) {
+                    continue;
+                }
+
+                $stringsMap = $this->mapStrings($fileId);
+
+                foreach ($catalogue->all() as $messages) {
+                    foreach ($messages as $id => $message) {
+                        if (!\array_key_exists($id, $stringsMap)) {
+                            continue;
+                        }
+
+                        $responses[] = $this->deleteString($stringsMap[$id]);
+                    }
+                }
+            }
+        }
+
+        foreach ($responses as $response) {
+            if (404 === $response->getStatusCode()) {
+                return;
```

Comment:
We will lose all next strings to delete here, we should use the logger to log a error (or a warning, as a not found string to delete could be reputed already deleted)

Reply:
Ok, seems legit to not log anything here